manifests/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -25,8 +25,8 @@ spec:
       maxUnavailable: {{ .Values.pilot.rollingMaxUnavailable }}
   selector:
     matchLabels:
-      app: istiod
       {{- if ne .Values.revision ""}}
```

Comment:
when and how does `.Values.revision` get initialized?

Reply:
and how does this work to fix the issue?

Reply:
The user sets it in the IOP. You cannot change label selector, its immutable. So with --force, it will delete the old one and create a new one, which breaks everything as noted in the k8s bug.